statics/js/components/graph-layout.js

```diff
@@ -1,6 +1,184 @@
+    },
+
+    setup: function(topology) {
+      this.topology = topology;
+    },
+
+    updateData: function(link) {
+      if (!link.target.metadata.LastUpdateMetric) {
+        return;
+      }
+
+      const defaultBandwidthBaseline = 1024 * 1024 * 1024; // 1 gbps
+
+      link.bandwidthBaseline = (this.topology.bandwidth.bandwidthThreshold === 'relative') ?
+        link.target.metadata.Speed || defaultBandwidthBaseline : 1;
+
+      link.bandwidthAbsolute = this.bandwidthFromMetrics(link.target.metadata.LastUpdateMetric);
+      link.bandwidth = link.bandwidthAbsolute / link.bandwidthBaseline;
+    },
+ 
+    hasData: function(link) {
+      if (!link.target.metadata.LastUpdateMetric) {
+        return false;
+      }
+
+      if (!link.bandwidth) {
+        return false;
+      }
+      return link.bandwidth > this.topology.bandwidth.active;
+    },
+
+    getText: function(link) {
+      return bandwidthToString(link.bandwidthAbsolute);
+    },
+
+    isActive: function(link) {
+      return (link.bandwidth > this.topology.bandwidth.active) && (link.bandwidth < this.topology.bandwidth.warning);
+    },
+
+    isWarning: function(link) {
+      return (link.bandwidth >= this.topology.bandwidth.warning) && (link.bandwidth < this.topology.bandwidth.alert);
+    },
+
+    isAlert: function(link) {
+      return link.bandwidth >= this.topology.bandwidth.alert;
+    },
+  },
+});
+
+var LinkLabelLatency = Vue.extend({
+  mixins: [apiMixin],
+
+  methods: {
+    setup: function(topology) {
+      this.active = 0;
+      this.warning = 10
+      this.alert = 100;
+    },
+
+    flowKey: function(a) {
+      return `TrackingID:${a.TrackingID},ANodeTID:${a.ANodeTID},BNodeTID:${a.BNodeTID}`;
+    },
+
+    flowCompare: function(a, b) {
+      const keya = this.flowKey(a);
+      const keyb = this.flowKey(b);
+      return keya.localeCompare(keyb);
+    },
+
+    updateLatency: function(link, flows) {
+      let prev, curr;
+      for (let i in flows) {
+        prev = curr;
+        curr = flows[i];
+
+        if (!prev) {
+          continue;
+        }
+
+        if (this.flowKey(curr) != this.flowKey(prev)) {
+          continue;
+        }
+
+        link.latencyTimestamp = Math.max(curr.Last, prev.Last);
+        link.latency = Math.abs(curr.RTT - prev.RTT) / 1000;
+        return true;
+      }
+      return false;
+    },
+
+    updateData: function(link) {
+      var self = this;
+
+      const a = link.source.metadata;
+      const b = link.target.metadata;
+
+      if (!a.Capture) {
+        return;
+      }
+      if (!b.Capture) {
+        return;
+      }
+
+      const limit = 10;
+      const anyNode = 'Within("'+a.TID+'", "'+b.TID+'")';
+      const query = 'G.Flows().Has("NodeTID", '+anyNode+').Has("RTT", NE(0)).Limit('+limit+')';
+
+      this.$topologyQuery(query)
+        .then(function(flows) {
+          if (flows.length === 0) {
+            return;
+          }
+          flows = flows.sort(self.flowCompare);
```

Comment:
you should do 2 gremlin request 
1st: flow = G.Flows().Has("NodeTID", '+anyNode+').Has("RTT", NE(0)).Limit(1)
2nd : flowNodes = G.Flows().Has("TrackingID", flow.TrackingID)

fill a hash with (foreach flowNodes)
RTTNodes[flowNodes.NoteTID] = flowNodes.RTT

pass the link, flow and the hash to your function updateLatency()

Your current implementation will not work in production, as it's currently bet that a same flow will be captured on nodeA and nodeB within your limit (10)

Reply:
But 1st step may return a flow which only passed via A (but not B) or B (but not A) - as a result we will not be able (in 2nd step) to get RTT data for both A and B.

Reply:
@hunchback  I would rather do a selection of the flows seen since a period (1 minute), so than we will not miss flow according to what Nicolas said and it will solve what you reported.

Reply:
As there is possible clock skew I have added Context("the last 5 minutes") to the query